src/main/java/org/alliancegenome/curation_api/services/helpers/diseaseAnnotations/WormDiseaseAnnotationCurie.java

```diff
@@ -28,4 +32,27 @@ public String getCurieID(String subject, String object, String reference, List<S
 		return super.getCurieID(subject, object, reference, null,null, null);
 	}
 
+	@Override
+	public String getCurieID(GeneDiseaseAnnotation annotation) {
+		return getDiseaseAnnotationCurieID(annotation.getSubject().getCurie(), annotation);
+	}
+
+	@Override
+	public String getCurieID(AlleleDiseaseAnnotation annotation) {
+		return getDiseaseAnnotationCurieID(annotation.getSubject().getCurie(), annotation);
+	}
+
+	@Override
+	public String getCurieID(AGMDiseaseAnnotation annotation) {
+		return getDiseaseAnnotationCurieID(annotation.getSubject().getCurie(), annotation);
+	}
+	
+	private String getDiseaseAnnotationCurieID(String subjectCurie, DiseaseAnnotation annotationDTO) {
```

Comment:
@markquintontulloch should probably pull DTO off the name here as its an entity and not a DTO.

Reply:
done